Provider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version11/GdsStatement.cs

```diff
@@ -210,11 +194,8 @@ protected override void Free(int option)
 			if (FreeNotNeeded(option))
 				return;
 
-			lock (_database.SyncObject)
-			{
-				DoFreePacket(option);
-				(Database as GdsDatabase).DeferredPackets.Enqueue(ProcessFreeResponse);
-			}
+			DoFreePacket(option);
+			(Database as GdsDatabase).DeferredPackets.Enqueue(ProcessFreeResponse);
```

Comment:
Database is guaranteed to be GdsDatabase here. I suggest several things:
1. Change as to cast. If my first statement ever becomes false, exception thrown here would be more informative: wrong cast vs nre.
2. Add ```GdsDatabase Database { get; set; }``` private(?) property, and use it everywhere inside the class, instead public interface property.